src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -60,6 +62,7 @@ export interface QueryEditorProps {
   filterBar?: any;
   prepend?: React.ComponentProps<typeof EuiCompressedFieldText>['prepend'];
   savedQueryManagement?: any;
+  additionalControls$?: BehaviorSubject<SearchBarControl[]>;
```

Comment:
Why do we need this to be an observable?

Reply:
Might not be a hard requirement now, but I was thinking, it might make sense to make it an observable so that the controls can be added on the fly, similar to dynamic action triggers.

Reply:
There are 2 problems I see with this approach.
1. Whats the usecase for such dynamic controls? Without a use I don't want to add more complexity to the query bar. It think it's more useful to pass the current app state to determine if the controls are needed rather than having an observable. 
2. Right now you can only add controls but not remove them. But this is an easy fix compared to the first one though